core/src/main/scala/kafka/message/InvalidMessageException.scala

```diff
@@ -17,9 +17,11 @@
 
 package kafka.message
 
+import org.apache.kafka.common.errors.CorruptRecordException
+
 /**
  * Indicates that a message failed its checksum and is corrupt
  */
-class InvalidMessageException(message: String) extends RuntimeException(message) {
+class InvalidMessageException(message: String) extends CorruptRecordException(message) {
```

Comment:
This is the only place I was required to do this. The reason is that both the clients and the server internals use Message.scala and ByteBufferMessageSet.scala. That means I can't change what those throw from InvalidMessageException. I want the server to be able to catch CorruptRecordException, without changing what the client is expected to catch.